File: src/main/java/com/safer/RH/models/Recrutement.java
package com.safer.RH.models;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;
import java.util.Collection;
import java.util.List;

@Entity
@Data
public class Recrutement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @OneToOne
    private Poste poste;
    @OneToMany(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private List<Candidature> candidats;
    @ManyToOne
    @JoinColumn(name = "recruteur_id", nullable = false)
    private Recruteur recruteur;
    private String typeContrat;
    @Enumerated(EnumType.STRING)
    private StateRecrutement statut;//(EN_COURS, TERMINE)
    private String etapeActuelle;
    private LocalDate dateDebut;
    private LocalDate dateFin;
}

File: src/app/api/models/recrutement.ts
import { Candidature } from "./Candidature";
import { Poste } from "./Poste";
import { Recruteur } from "./recruteur";

export interface Recrutement {
    id?: number;
    poste?:Poste;
    candidats?:Candidature[];
    recruteur?: Recruteur;
    statut?:string;
    dateDebut?: string;
    dateFin?: string;
    
}

export interface RecrutementDto {
    id?:number;
    poste?:string;
    salaire?:number;
    niveauEtude?:string;
    recruteur?:string;
    statut?:string;
    etapeActuelle?:string;
    dateDebut?:string;
    dateFin?:string;
}